packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(unused_imports)]
+
+use core::convert::Infallible;
+
+use packable::{
+    error::{UnknownTagError, UnpackError, UnpackErrorExt},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+#[derive(Debug)]
+pub struct PickyError(u8);
+
+impl From<Infallible> for PickyError {
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+fn verify<const VERIFY: bool>(&value: &u64, _: &()) -> Result<(), PickyError> {
+    if !VERIFY || value == 42 {
+        Ok(())
+    } else {
+        Err(PickyError(value as u8))
```

Comment:
Why is the value accepted as u64 if it needs to be an u8 at the end?

Reply:
Copy paste error

Reply:
Ah no, not copy paste error. This is a failing test, it's supposed to be a wrong type. I've put u64 but I could have put Vec, it' just there to make it fail.

Reply:
So providing a wrong type will fail to compile already, right?

Reply:
Yes